lib/spack/spack/test/cmd/load.py

```diff
@@ -51,21 +58,33 @@ def test_load(install_mockery, mock_fetch, mock_archive, mock_packages):
     print(install_out)
     mpileaks_spec = spack.spec.Spec("mpileaks").concretized()
 
-    sh_out = load("--sh", "--only", "package", "mpileaks")
-    csh_out = load("--csh", "--only", "package", "mpileaks")
+    if sys.platform == "win32":
+        bat_out = load("--bat", "--only", "package", "mpileaks")
 
-    # Test prefix inspections
-    sh_out_test = "export CMAKE_PREFIX_PATH=%s" % mpileaks_spec.prefix
-    csh_out_test = "setenv CMAKE_PREFIX_PATH %s" % mpileaks_spec.prefix
-    assert sh_out_test in sh_out
-    assert csh_out_test in csh_out
+        # Test prefix inspections
+        bat_out_test = 'set "CMAKE_PREFIX_PATH=%s"' % mpileaks_spec.prefix
+        assert bat_out_test in bat_out
 
-    # Test hashes recorded properly
-    hash_test_replacements = (uenv.spack_loaded_hashes_var, mpileaks_spec.dag_hash())
-    sh_hash_test = "export %s=%s" % hash_test_replacements
-    csh_hash_test = "setenv %s %s" % hash_test_replacements
-    assert sh_hash_test in sh_out
-    assert csh_hash_test in csh_out
+        # Test hashes recorded properly
+        hash_test_replacements = (uenv.spack_loaded_hashes_var, mpileaks_spec.dag_hash())
+        bat_hash_test = 'set "%s=%s"' % hash_test_replacements
+        assert bat_hash_test in bat_out
+    else:
+        sh_out = load("--sh", "--only", "package", "mpileaks")
+        csh_out = load("--csh", "--only", "package", "mpileaks")
+
+        # Test prefix inspections
+        sh_out_test = "export CMAKE_PREFIX_PATH=%s" % mpileaks_spec.prefix
+        csh_out_test = "setenv CMAKE_PREFIX_PATH %s" % mpileaks_spec.prefix
+        assert sh_out_test in sh_out
+        assert csh_out_test in csh_out
+
+        # Test hashes recorded properly
+        hash_test_replacements = (uenv.spack_loaded_hashes_var, mpileaks_spec.dag_hash())
+        sh_hash_test = "export %s=%s" % hash_test_replacements
+        csh_hash_test = "setenv %s %s" % hash_test_replacements
+        assert sh_hash_test in sh_out
+        assert csh_hash_test in csh_out
```

Comment:
Like `test_manpath_trailing_colon `, we don't have to only execute `--sh` on windows

Unlike the `test_manpath_trailing_colon`, I think this could be consolidated with parameterization:

```
@pytest.parameter(shell, [
  (--bat, 'set "%s=%s", X),
  (--sh, "export CMAKE_PREFIX_PATH=%s", "setenv %s %s"), 
  (--csh, ...)
])
``` 

assuming there is a sensible setting for X (I assume on Windows we also set `uenv.spack_loaded_hashes_var`).

Assuming that makes sense, can you look into whether a similar consolidation can be applied to the other tests (I don't think `test_manpath_trailing_colon` could be updated easily in this way, so that might be the case elsewhere).